train/bleu.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+public class BLEU
+{
+    public static int str1_length;
+    public static int str2_length;
+    public class Ngram
+    {
+        private int N;
+        private string OriginalSentence;
+        public Ngram(string sentence, int n)
+        {
```

Comment:
We usually want to validate the constructor's parameters here. e.g.
if n is a positive integer?
if sentence is not null?